src/openag_brain/software_modules/recipe_handler.py

```diff
@@ -105,6 +108,43 @@ def __iter__(self):
             yield (time.time(), variable, value)
         yield (time.time(), RECIPE_END.name, self.id)
 
+def hrs_to_seconds(hrs):
+    return hrs * (60 * 60)
+
+def days_to_seconds(days):
+    return hrs_to_seconds(days * 24)
+
+@interpret_recipe.register("phased")
+class PhasedRecipeInterpreter:
+    def __init__(self, recipe, start_time=None, timeout=1):
+        init_time = rospy.get_time()
+        start_time = start_time or init_time
+        if start_time > init_time:
+            raise ValueError("Recipes cannot be scheduled for the future")
+        self.start_time = start_time
+        self.stages = recipe["operations"]
+        self.id = recipe["_id"]
+        self.timeout = timeout
+
+    def __iter__(self):
+        yield (rospy.get_time(), RECIPE_START.name, self.id)
+        for stage in self.stages:
+            stage_duration = days_to_seconds(stage["days"])
+            end_of_stage = (rospy.get_time() - self.start_time) + stage_duration
```

Comment:
@LeonChambers ah, I'm remembering why I wrote it this way.

Imagine you have a 19hr stage with a 12hr night and a 12hr day. What should happen?

You could enforce the stage duration and truncate the last phase:

     stage |-------------------|
     phase |------------|------|

Or you could allow the phase to run its course and overrun the stage time

     stage |-------------------|
     phase |------------|------------|

I prefer the latter, because you can imagine a situation where a night is cut short, followed by an intense day in the next stage and the plant ends up burned. So overrunning the stage seems like the most benign solution... a sensible default.

Of course, this implementation doesn't solve everything. One could also imagine the following mistake happening:

    |day|night|day|night|day|
                            |day|night|day|night|

But I didn't bother to guard against it. Maybe we should!

Bottom line, I think it makes more sense to let the stage time drift long if it means completing phases correctly, since I think this will be a better default for growth outcomes.

Thoughts?